firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvents.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import com.google.firebase.encoders.DataEncoder
+import com.google.firebase.encoders.FieldDescriptor
+import com.google.firebase.encoders.ObjectEncoderContext
+import com.google.firebase.encoders.json.JsonDataEncoderBuilder
+
+/**
+ * Contains functions for [SessionEvent]s.
+ *
+ * @hide
+ */
+internal object SessionEvents {
+  /** JSON [DataEncoder] for [SessionEvent]s. */
+  // TODO(mrober): Replace with firebase-encoders-processor when it can encode Kotlin data classes.
+  internal val SESSION_EVENT_ENCODER: DataEncoder =
+    JsonDataEncoderBuilder()
+      .configureWith {
```

Comment:
How does it work for hierarchical fields? Are these encoders just for the KotlinData classes? As we encode the fields, they use these encoders to package the fields?

Reply:
There are encoders for every type we need to encode. This will be replaced by an annotation processor that will generate this for us after Acore updates firebase-encoders-processor.